PracticeGit/Program.c

```diff
@@ -1,13 +1,44 @@
 // ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
-// ------------------------------------------------------------------
-// Program.c
 // ------------------------------------------------------------------------------------------------
-
-#include <stdio.h>
+// Program.c - Program that has quick sort and binary search function
+// ------------------------------------------------------------------------------------------------
 #include "Program.h"
 
-void PrintMessage () {
-   printf ("Hey!! World\n");
+#define ELEMENT_NOT_FOUND -1
+
+void Swap (int* a, int* b) {
+   if (a != b) {   // swap only if it is in diferent addresses
+      *a = *a + *b;
+      *b = *a - *b;
+      *a = *a - *b;
+   }
+}
+
+int Partition (int array[], int left, int right) {
```

Comment:
Similarly, the naming for this method can be better. Also, this is used only once and can very well be written inside the QuickSort () function itself with meaningful comments.